libtransmission/bandwidth.h

```diff
@@ -110,8 +110,45 @@ struct tr_bandwidth
     tr_priority_t priority;
     int magicNumber;
     unsigned int uniqueKey;
-    tr_ptrArray children; /* struct tr_bandwidth */
+    tr_ptrArray children; /* struct tr_bandwidth, TODO: replace with std::vector<tr_bandwidth> */
     struct tr_peerIo* peer;
+
+    /**
+     * @brief clamps byteCount down to a number that this bandwidth will allow to be consumed
+    */
+    [[nodiscard]] unsigned int clamp(tr_direction dir, unsigned int byteCount) const;
+
+    void setPeer(tr_peerIo* peer);
+
+    /** @brief Get the raw total of bytes read or sent by this bandwidth subtree. */
+    unsigned int getRawSpeed_Bps(uint64_t const now, tr_direction const dir) const;
+
+    /** @brief Get the number of piece data bytes read or sent by this bandwidth subtree. */
+    unsigned int getPieceSpeed_Bps(uint64_t const now, tr_direction const dir) const;
+
+    /**
+     * @brief Notify the bandwidth object that some of its allocated bandwidth has been consumed.
+     * This is is usually invoked by the peer-io after a read or write.
+     */
+    void used(tr_direction dir, size_t byteCount, bool isPieceData, uint64_t now);
```

Comment:
I feel like the previous name `tr_bandwidthUsed()` was not very descriptive and this just makes the pre-existing problem even more obvious. Maybe this should be replaced with a more descriptive name.